src/transformers/mocha.test.ts

```diff
@@ -162,3 +162,79 @@ describe('suite', () => {
 })
     `
 )
+
+testChanged(
+  'transforms this',
+  `
+// @flow
+describe('describe', function () {
+  before(function() {
+    this.hello = 'hi';
+  });
+  afterAll(function () {
+    console.log(this.hello);
+  });
+  beforeEach(function () {
+    this.goodbye = 'bye';
+  });
+  afterEach(function () {
+    console.log(this.hello);
+  });
+
+  before('some text', () => {});
+  after('some text', () => {});
+  beforeEach('some text', () => {});
+  afterEach('some text', () => {});
+
+  context('context', () => {
+    it('it', function () {
+      console.log(this.hello);
+      console.log(this.goodbye);
+    });
+    specify('specify', function () {
+      console.log(this.hello);
+      console.log(this.goodbye);
+    })
+  })
+})
+`,
+  `
+// @flow
+describe('describe', () => {
+  let testContext;
+
+  before(() => {
```

Comment:
Oops, I intended for this to be a **`beforeAll`**, not a **`before`** (that's probably where some of this confusion is from).

Reply:
Yes. :)